core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala

```diff
@@ -70,6 +71,17 @@ private[ui] class StagePage(parent: StagesTab) extends WebUIPage("stage") {
 
   private val displayPeakExecutionMemory = parent.conf.getBoolean("spark.sql.unsafe.enabled", true)
 
+  private def getLocalitySummaryString(stageData: StageUIData): String = {
+    val localityCounts = new HashMap[String, Long]
+    stageData.taskData.values.foreach( taskUIData => {
+      val locality = taskUIData.taskInfo.taskLocality.toString
+      localityCounts.put(locality, localityCounts.getOrElse(locality, 0L) + 1)
+    })
+    return localityCounts.map { _ match {
```

Comment:
Correct Sean. Thanks ! Actually, I'm implementing second Kay's suggestion (using "Advanced Metrics" checkbox). I will update the PR.